src/promptflow-index/requirements.txt

```diff
@@ -0,0 +1,3 @@
+azureml-rag[azure,faiss,cognitive_search,document_parsing]
+azure-ai-ml>=1.14.0
```

Comment:
not sure if we should separate azure depedency to extra?

Reply:
Isn't promptflow-index supposed to be "run locally" without AzureML dependency?
If it takes azure-ai-ml dependency anyway, maybe we should have the local index building in azure-ai-ml as well ...

Reply:
@danielsc thoughts?

Reply:
My bad, azure-ai-ml is not needed. Will remove it

Reply:
I don't think simply remove azure-ai-ml dependency works, azureml-rag[azure] still depends on azure-ai-ml. And from what I read, most of the index built method rely on Azure? _create_mlindex_from_existing_ai_search need to leverage mlclient to get connection, another path create_index_from_raw_embeddings totally rely on ACS to create an MLIndex.